src/components/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,44 @@
+import { Content, Text } from './Sidebar.styled';
+import { useState, useEffect } from 'react';
+import categoriesData from '../../mocks/en-us/product-categories.json';
+
+const Sidebar = ({ filterProducts }) => {
+
+    const [categories, setCategories] = useState(categoriesData.results.map(category => ({...category, selected: true })));
+    
+    const onCategorySelected = (event) => {
+        const id = event.target.getAttribute('name');
+
+        const results = categories.map(c => {
+            if (id === c.id) {
+                c.selected = !c.selected;
+            }
+            return c;
+        });
+
+        setCategories(results);
+    }
+
+    useEffect(() => {
+        filterProducts(categories.filter(category => category.selected).map(category => category.data.name.toLowerCase()));
+    }, [categories, filterProducts]);
+
+    return (
+        <Content>
+            {
+                categories.map(category =>(
+                    <Text 
+                        onClick={onCategorySelected}
```

Comment:
you could do something like this to avoid having to access the target:

`onClick={() => onCategorySelected(category.id)}`